drivers/sensor/vcnl4040/vcnl4040.c

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) 2020 Richard Osterloh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT vishay_vcnl4040
+
+#include "vcnl4040.h"
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+#include <logging/log.h>
+#include <stdlib.h>
+
+LOG_MODULE_REGISTER(vcnl4040, CONFIG_SENSOR_LOG_LEVEL);
+
+int vcnl4040_read(const struct device *dev, uint8_t reg, uint16_t *out)
+{
+	const struct vcnl4040_config *config = dev->config;
+	struct vcnl4040_data *data = dev->data;
+	uint8_t buff[2] = { 0 };
+	int ret = 0;
+
+	ret = i2c_write_read(data->i2c, config->i2c_address,
+			     &reg, sizeof(reg), buff, sizeof(buff));
+
+	*out = sys_get_le16(buff);
```

Comment:
Is it required to set output content even on failure?

Reply:
Not needed in this case